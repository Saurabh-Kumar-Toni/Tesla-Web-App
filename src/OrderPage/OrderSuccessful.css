.orderContainer{
    height: auto;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    overflow: scroll;
}

.orderCompleteSection{
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    margin-top: 50px;
}
.text1{
    text-align: center;
    font-weight: normal;
   font-size: 36px;
   line-height: 45px;
   color: #000000;
}

.text2{
    font-weight: normal;
font-size: 48px;
line-height: 60px;
color: #000000;
}

.summarySection{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 100%;

}

.footerSection{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}


.featuresSection{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 100%;

}
.featuresItem{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    width: 350px;
    padding: 20px;
}
