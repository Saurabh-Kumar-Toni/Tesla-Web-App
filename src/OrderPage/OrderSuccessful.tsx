import React from 'react';
import RedCar from '../Images/red.png';
import CheckMark from '../Images/greentick.png';
import SuccessImage from '../Images/mouse.png';
import Button from '../Components/Button';
import './OrderSuccessful.css';
const displayName = 'Model S';
const variantData = [
    {
        acceleration: { speed: "0-60", time: 1.99 },
        range: '320',
        topSpeed: '250'
    }
]
const selectedVariant = [
    {
        variantName:'Plaid+',
        price:149990
    }
];
const selectedPaint =[
    {
        name:'Red',
        price:2500
    }
]
const selectedWheels=[
    {
        name:'Tempest 19” Wheels',
        price:0,
        src: "https://static-assets.tesla.com/share/tesla_design_studio_assets/MODEL3/UI/option-wheels-20-uberturbine.png?&version=v0028d202104080411"
    }
]
const selectedInteriors=[
    {
        name:'',
        price:0
    }
]

const selectedSelfDriving=[{
    default:true,
    price:10000
}]
const addSelfBool = true;

const totalPrice = selectedVariant[0].price+selectedPaint[0].price+
selectedWheels[0].price+selectedInteriors[0].price+selectedSelfDriving[0].price;

const OrderSuccessful: React.FC = () => {
    return (<main className="orderContainer">
        <section className="orderCompleteSection">
            <img src={CheckMark} height={'52px'} width={'52px'} alt="done img" />
            <h1 className="text1">Your Order is Complete</h1>
        </section>
        <section>
            <img src={SuccessImage} height={'323px'} width={'306px'} alt='success img' />
        </section>

        <section className="summarySection" style={{ width: '375px' }}>

            <div className='accSection'>
                <h4 className='text2'>{displayName}</h4>
                {
                    variantData?.map((item: any, id = 0) => {
                        return (<div className="accDetails" key={id}>
                            <div className="accItem">
                                <p>{item.range + ' mi'}</p>
                                <p>Est Range</p>
                            </div>
                            <div className="accItem">
                                <p>{item.topSpeed + ' mph'}</p>
                                <p>Top Speed</p>
                            </div>
                            <div className="accItem">
                                <p>{item.acceleration.time}</p>
                                <p>{item.acceleration.speed}</p>
                            </div>


                        </div>)
                    })
                }
            </div>
            <div>
                <img src={RedCar} width="100%" />
            </div>
            <div className="buttonsSection">
        

                    {
                        selectedVariant.map((item: any, id = 0) => {
                            return (
                                <Button key={id} text={item.variantName + "\t $" + item.price} className={'configBtn'} onClick={() =>{}} />
                            )
                        })
                    }
            </div>
        </section>

        <section className="featuresSection"> 

                    <div className="featuresItem">
                        <div className="accItem">
                        <p>{selectedPaint[0]?.name + ' Multi-Coat'}</p>
                        <p>{selectedPaint[0]?.price === 0 ? 'Included' :
                            '$ ' + selectedPaint[0]?.price}</p>
                        </div>
                        <div className={`paint ${selectedPaint[0].name.split(" ")[0]}`}  />
                    </div>
                    <div className="featuresItem">
                        <div className="accItem">
                        <p>{selectedWheels[0]?.name}</p>
                        <p>{selectedWheels[0]?.price === 0 ? 'Included' :
                            '$ ' + selectedPaint[0]?.price}</p>
                        </div>
                        <div>
                        <img src={selectedWheels[0].src}
                                        className="wheelsImg"
                                        alt={selectedWheels[0].name}
                                        onClick={() =>{}} />
                            </div>
                    </div>
                    <div className="featuresItem">
                        <div className="accItem">
                        <p>{selectedInteriors[0]?.name ===''?'All Black Decor':selectedInteriors[0].name}</p>
                        <p>{selectedInteriors[0]?.price === 0 ? 'Included' :
                            '$ ' + selectedInteriors[0]?.price}</p>
                        </div>
                        <div id={selectedPaint[0].name.split(" ")[0]} className={`paint`}  />
                      
                    </div>
                    {
                        addSelfBool?
                        <div className="featuresItem">
                             <div className="accItem">
                                 <p>Full self-driving capabilities</p>
                                 <p>$ {selectedSelfDriving[0].price}</p>
                             </div>   
                        </div>:''
                    }

        </section>
        {/* Price Section */}
        <section className="footerSection">
            <Button text={'$ ' + totalPrice.toString().replace(/\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g, ",")} 
            onClick={()=>{}}
            className="configOrderBtn bg-light"/>

            <div className="buttonContainer" >
            <Button text={'Explore Model S'} 
            onClick={()=>{}}
            className="configOrderBtn"/>
              <Button text={'Download Invoice'} 
            onClick={()=>{}}
            className="configOrderBtn bg-dark"/>
            </div>
        </section>


    </main>)
}
export default OrderSuccessful;