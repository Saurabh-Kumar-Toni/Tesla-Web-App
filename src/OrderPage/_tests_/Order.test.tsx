import { render, screen } from '@testing-library/react';
import OrderSuccessful from '../OrderSuccessful';

test('check if the Order page is rendered properly or not', () => {
 
    const {container} = render(<OrderSuccessful />)
    expect(container.firstChild).toMatchSnapshot();
  });