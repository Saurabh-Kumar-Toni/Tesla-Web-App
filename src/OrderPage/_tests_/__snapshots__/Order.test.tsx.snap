// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if the Error page is rendered properly or not 1`] = `
<main
  class="orderContainer"
>
  <section
    class="orderCompleteSection"
  >
    <img
      alt="done img"
      height="52px"
      src="greentick.png"
      width="52px"
    />
    <h1
      class="text1"
    >
      Your Order is Complete
    </h1>
  </section>
  <section>
    <img
      alt="success img"
      height="323px"
      src="mouse.png"
      width="306px"
    />
  </section>
  <section
    class="summarySection"
    style="width: 375px;"
  >
    <div
      class="accSection"
    >
      <h4
        class="text2"
      >
        Model S
      </h4>
      <div
        class="accDetails"
      >
        <div
          class="accItem"
        >
          <p>
            320 mi
          </p>
          <p>
            Est Range
          </p>
        </div>
        <div
          class="accItem"
        >
          <p>
            250 mph
          </p>
          <p>
            Top Speed
          </p>
        </div>
        <div
          class="accItem"
        >
          <p>
            1.99
          </p>
          <p>
            0-60
          </p>
        </div>
      </div>
    </div>
    <div>
      <img
        src="red.png"
        width="100%"
      />
    </div>
    <div
      class="buttonsSection"
    >
      <button
        class="configBtn"
      >
        Plaid+	 $149990
      </button>
    </div>
  </section>
  <section
    class="featuresSection"
  >
    <div
      class="featuresItem"
    >
      <div
        class="accItem"
      >
        <p>
          Red Multi-Coat
        </p>
        <p>
          $ 2500
        </p>
      </div>
      <div
        class="paint Red"
      />
    </div>
    <div
      class="featuresItem"
    >
      <div
        class="accItem"
      >
        <p>
          Tempest 19” Wheels
        </p>
        <p>
          Included
        </p>
      </div>
      <div>
        <img
          alt="Tempest 19” Wheels"
          class="wheelsImg"
          src="https://static-assets.tesla.com/share/tesla_design_studio_assets/MODEL3/UI/option-wheels-20-uberturbine.png?&version=v0028d202104080411"
        />
      </div>
    </div>
    <div
      class="featuresItem"
    >
      <div
        class="accItem"
      >
        <p>
          All Black Decor
        </p>
        <p>
          Included
        </p>
      </div>
      <div
        class="paint"
        id="Red"
      />
    </div>
    <div
      class="featuresItem"
    >
      <div
        class="accItem"
      >
        <p>
          Full self-driving capabilities
        </p>
        <p>
          $ 
          10000
        </p>
      </div>
    </div>
  </section>
  <section
    class="footerSection"
  >
    <button
      class="configOrderBtn bg-light"
    >
      $ 162,490
    </button>
    <div
      class="buttonContainer"
    >
      <button
        class="configOrderBtn"
      >
        Explore Model S
      </button>
      <button
        class="configOrderBtn bg-dark"
      >
        Download Invoice
      </button>
    </div>
  </section>
</main>
`;

exports[`check if the Order page is rendered properly or not 1`] = `
<main
  class="orderContainer"
>
  <section
    class="orderCompleteSection"
  >
    <img
      alt="done img"
      height="52px"
      src="greentick.png"
      width="52px"
    />
    <h1
      class="text1"
    >
      Your Order is Complete
    </h1>
  </section>
  <section>
    <img
      alt="success img"
      height="323px"
      src="mouse.png"
      width="306px"
    />
  </section>
  <section
    class="summarySection"
    style="width: 375px;"
  >
    <div
      class="accSection"
    >
      <h4
        class="text2"
      >
        Model S
      </h4>
      <div
        class="accDetails"
      >
        <div
          class="accItem"
        >
          <p>
            320 mi
          </p>
          <p>
            Est Range
          </p>
        </div>
        <div
          class="accItem"
        >
          <p>
            250 mph
          </p>
          <p>
            Top Speed
          </p>
        </div>
        <div
          class="accItem"
        >
          <p>
            1.99
          </p>
          <p>
            0-60
          </p>
        </div>
      </div>
    </div>
    <div>
      <img
        src="red.png"
        width="100%"
      />
    </div>
    <div
      class="buttonsSection"
    >
      <button
        class="configBtn"
      >
        Plaid+	 $149990
      </button>
    </div>
  </section>
  <section
    class="featuresSection"
  >
    <div
      class="featuresItem"
    >
      <div
        class="accItem"
      >
        <p>
          Red Multi-Coat
        </p>
        <p>
          $ 2500
        </p>
      </div>
      <div
        class="paint Red"
      />
    </div>
    <div
      class="featuresItem"
    >
      <div
        class="accItem"
      >
        <p>
          Tempest 19” Wheels
        </p>
        <p>
          Included
        </p>
      </div>
      <div>
        <img
          alt="Tempest 19” Wheels"
          class="wheelsImg"
          src="https://static-assets.tesla.com/share/tesla_design_studio_assets/MODEL3/UI/option-wheels-20-uberturbine.png?&version=v0028d202104080411"
        />
      </div>
    </div>
    <div
      class="featuresItem"
    >
      <div
        class="accItem"
      >
        <p>
          All Black Decor
        </p>
        <p>
          Included
        </p>
      </div>
      <div
        class="paint"
        id="Red"
      />
    </div>
    <div
      class="featuresItem"
    >
      <div
        class="accItem"
      >
        <p>
          Full self-driving capabilities
        </p>
        <p>
          $ 
          10000
        </p>
      </div>
    </div>
  </section>
  <section
    class="footerSection"
  >
    <button
      class="configOrderBtn bg-light"
    >
      $ 162,490
    </button>
    <div
      class="buttonContainer"
    >
      <button
        class="configOrderBtn"
      >
        Explore Model S
      </button>
      <button
        class="configOrderBtn bg-dark"
      >
        Download Invoice
      </button>
    </div>
  </section>
</main>
`;
